File: bot-base-client/src/main/java/gg/mineral/bot/base/lwjgl/opengl/GL30.java
package gg.mineral.bot.base.lwjgl.opengl;

import gg.mineral.bot.impl.config.BotGlobalConfig;

public class GL30 {

    public static void glBindFramebuffer(int target, int framebuffer) {
        if (BotGlobalConfig.isHeadless())
            return;

        org.lwjgl.opengl.GL30.glBindFramebuffer(target, framebuffer);
    }

    public static void glBindRenderbuffer(int target, int renderbuffer) {
        if (BotGlobalConfig.isHeadless())
            return;

        org.lwjgl.opengl.GL30.glBindRenderbuffer(target, renderbuffer);
    }

    public static void glDeleteRenderbuffers(int renderbuffer) {
        if (BotGlobalConfig.isHeadless())
            return;

        org.lwjgl.opengl.GL30.glDeleteRenderbuffers(renderbuffer);
    }

    public static void glDeleteFramebuffers(int framebuffer) {
        if (BotGlobalConfig.isHeadless())
            return;

        org.lwjgl.opengl.GL30.glDeleteFramebuffers(framebuffer);
    }

    public static int glGenFramebuffers() {
        if (BotGlobalConfig.isHeadless())
            return 0;

        return org.lwjgl.opengl.GL30.glGenFramebuffers();
    }

    public static int glGenRenderbuffers() {
        if (BotGlobalConfig.isHeadless())
            return 0;

        return org.lwjgl.opengl.GL30.glGenRenderbuffers();
    }

    public static void glRenderbufferStorage(int target, int internalformat, int width, int height) {
        if (BotGlobalConfig.isHeadless())
            return;

        org.lwjgl.opengl.GL30.glRenderbufferStorage(target, internalformat, width, height);
    }

    public static void glFramebufferRenderbuffer(int target, int attachment, int renderbuffertarget, int renderbuffer) {
        if (BotGlobalConfig.isHeadless())
            return;

        org.lwjgl.opengl.GL30.glFramebufferRenderbuffer(target, attachment, renderbuffertarget, renderbuffer);
    }

    public static int glCheckFramebufferStatus(int target) {
        if (BotGlobalConfig.isHeadless())
            return 36053;

        return org.lwjgl.opengl.GL30.glCheckFramebufferStatus(target);
    }

    public static void glFramebufferTexture2D(int target, int attachment, int textarget, int texture, int level) {
        if (BotGlobalConfig.isHeadless())
            return;

        org.lwjgl.opengl.GL30.glFramebufferTexture2D(target, attachment, textarget, texture, level);
    }

}

File: bot-lwjgl-bridge/src/main/kotlin/gg/mineral/bot/lwjgl/opengl/GL30.kt
package gg.mineral.bot.lwjgl.opengl

import gg.mineral.bot.impl.config.BotGlobalConfig
import org.lwjgl.opengl.GL30

object GL30 {
    @JvmStatic
    fun glBindFramebuffer(target: Int, framebuffer: Int) {
        if (BotGlobalConfig.headless) return

        GL30.glBindFramebuffer(target, framebuffer)
    }

    @JvmStatic
    fun glBindRenderbuffer(target: Int, renderbuffer: Int) {
        if (BotGlobalConfig.headless) return

        GL30.glBindRenderbuffer(target, renderbuffer)
    }

    @JvmStatic
    fun glDeleteRenderbuffers(renderbuffer: Int) {
        if (BotGlobalConfig.headless) return

        GL30.glDeleteRenderbuffers(renderbuffer)
    }

    @JvmStatic
    fun glDeleteFramebuffers(framebuffer: Int) {
        if (BotGlobalConfig.headless) return

        GL30.glDeleteFramebuffers(framebuffer)
    }

    @JvmStatic
    fun glGenFramebuffers(): Int {
        if (BotGlobalConfig.headless) return 0

        return GL30.glGenFramebuffers()
    }

    @JvmStatic
    fun glGenRenderbuffers(): Int {
        if (BotGlobalConfig.headless) return 0

        return GL30.glGenRenderbuffers()
    }

    @JvmStatic
    fun glRenderbufferStorage(target: Int, internalformat: Int, width: Int, height: Int) {
        if (BotGlobalConfig.headless) return

        GL30.glRenderbufferStorage(target, internalformat, width, height)
    }

    @JvmStatic
    fun glFramebufferRenderbuffer(target: Int, attachment: Int, renderbuffertarget: Int, renderbuffer: Int) {
        if (BotGlobalConfig.headless) return

        GL30.glFramebufferRenderbuffer(target, attachment, renderbuffertarget, renderbuffer)
    }

    @JvmStatic
    fun glCheckFramebufferStatus(target: Int): Int {
        if (BotGlobalConfig.headless) return 36053

        return GL30.glCheckFramebufferStatus(target)
    }

    @JvmStatic
    fun glFramebufferTexture2D(target: Int, attachment: Int, textarget: Int, texture: Int, level: Int) {
        if (BotGlobalConfig.headless) return

        GL30.glFramebufferTexture2D(target, attachment, textarget, texture, level)
    }
}
